File: Strings/ReverseString.php
<?php

class Solution {

    /**
     * @param String[] $s
     * @return NULL
     */
    function reverseString(&$s) {
        $start = 0;
        $end = count($s) - 1;

        while ($start < $end) {
            $tmp = $s[$start];
            $s[$start] = $s[$end];
            $s[$end] = $tmp;
            
            $start++;
            $end--;
        }
    }
}

File: Recursion1/reverseString.py
class Solution:
    def reverseString(self, s: list[str], start = None, end = None) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        if start == None:
            start = 0
        
        if end == None:
            end = len(s) - 1

        if start >= end:
            return
        
        s[start], s[end] = s[end], s[start]
        self.reverseString(s, start + 1, end - 1)